### PR TITLE
Core actions feature branch

### PR DESCRIPTION
**Pull request by CarlosNZ**
Includes #976 #978 #980 

Ready to merge into develop once we're happy that it's stable.

Front-end: https://github.com/openmsupply/conforma-web-app/pull/1455

### CODE REVIEW

File: database/buildSchema/43_views_functions_triggers.sql [hunk 1]
```diff
@@ -286,6 +286,20 @@ $$
...
+    SELECT
+        parameters::text
+    FROM
+        public.template_element
+    WHERE
+        id = $1.id
+$$
+LANGUAGE sql
+STABLE;
+
```
Comment (CarlosNZ) [inline database/buildSchema/43_views_functions_triggers.sql:302]: @nmadruga This was the missing function I mentioned in the chat.

**Review by nmadruga — COMMENTED** [review]
Oh - I just noticed now that I forgot to submit this review : )

File: src/components/actions/coreActions.ts [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+    // other cases for incrementing stage must be specified in template actions.
+    {
+      code: 'incrementStage',
+      path: '../plugins/action_increment_stage/src/index.ts',
+      name: 'Increment Stage',
+      trigger: 'ON_REVIEW_SUBMIT',
+      event_code: null,
+      sequence: -3,
+      condition: {
+        operator: 'AND',
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/components/actions/coreActions.ts:382]: Okay, thanks. That code block has JSON errors in it though. Would you mind having another look at it (or pointing me to where you got it and I'll copy it over)?

[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/components/actions/coreActions.ts:382]: Okay, I got it from Angola ProdReg: 1a7f4b1b

File: src/components/actions/coreActions.ts [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+              {
+                operator: 'objectProperties',
+                children: ['applicationData.reviewData.latestDecision.decision'],
+              },
+              'CONFORM',
+            ],
+          },
+          // We also want to make sure that it won't move to the next stage if
+          // the reviewer is not reviewing the whole application (NEEDS
+          // RE-WORDING)
```
Comment (CarlosNZ) [inline src/components/actions/coreActions.ts:400]: I'll get you to help re-word this better when we go through it together @nmadruga

**Review by nmadruga — APPROVED** [review]
Peer reviewed this by Carl and I. 

👍

**Pull request merged**